src/D2L.CodeStyle.Analyzers/Immutability/UnsafeStaticsAnalyzer.cs

```diff
@@ -143,44 +139,36 @@ MutabilityInspector inspector
 			ITypeSymbol fieldOrPropertyType,
 			string fieldOrPropertyName
 		) {
+			bool hasUnauditedAnnotation = Attributes.Statics.Unaudited
+				.IsDefined( fieldOrProperty );
 
-			var diagnostics = GatherDiagnostics(
-				inspector,
-				fieldOrProperty: fieldOrProperty,
-				location: location,
-				isStatic: isStatic,
-				fieldOrPropertyType: fieldOrPropertyType,
-				fieldOrPropertyName: fieldOrPropertyName
-			);
+			bool hasAuditedAnnotation = Attributes.Statics.Audited
+				.IsDefined( fieldOrProperty );
 
-			// We're manually using enumerators here.
-			// - if we used IEnumerable directly we'd re-compute the first
-			//   diagnostic in the GatherDiagnostics generator
-			// - if we did .ToArray() early then we would avoid multiple
-			//   enumeration but would compute diagnostics even when we
-			//   ultimately ignore them due to annotations
-			using( var diagnosticsEnumerator = diagnostics.GetEnumerator() ) {
-				ProcessDiagnostics(
-					context,
-					diagnosticsEnumerator,
-					fieldOrProperty,
-					location,
-					fieldOrPropertyName
+			bool hasAnnotation = hasUnauditedAnnotation || hasAuditedAnnotation;
+
+			if ( !isStatic && hasAnnotation ) {
+				// non-statics don't need these annotations.
+				// TODO: Consider dropping the static-specific annotations in
+				// favour of Mutability.Audited/Unaudited. Will require some
+				// effort due to the reporting we do but it'll make things
+				// simpler once we merge the analyzers.

```

Comment:
Once you get the unnecessary diagnostics working for `Mutability.*` attributes I'll investigate this. The benefit would be simplifying the code (which is more obvious when considering how to merge the analyzers.)